From 58a2e17aa66340ca4cce3b962d90e54bd32652e0 Mon Sep 17 00:00:00 2001
From: Josua Mayer <josua@solid-run.com>
Date: Tue, 7 Apr 2026 17:32:46 +0300
Subject: [PATCH] phy: lynx-28g: fix of_node reference counting

The for_each_available_child_of_node helper function calls of_node_put
automatically on each iteration.

Remove the duplicate call from lynx_28g_get_lanephy_node_optional,
breaking an infinite loop observed in certain kernel configuration.

Fixes: "phy: lynx-28g: support overriding per-lane equalization parameters"
Signed-off-by: Josua Mayer <josua@solid-run.com>
---
 drivers/phy/freescale/phy-fsl-lynx-28g.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/drivers/phy/freescale/phy-fsl-lynx-28g.c b/drivers/phy/freescale/phy-fsl-lynx-28g.c
index 583156e67aa1..f8d21d686206 100644
--- a/drivers/phy/freescale/phy-fsl-lynx-28g.c
+++ b/drivers/phy/freescale/phy-fsl-lynx-28g.c
@@ -787,8 +787,6 @@ static struct device_node *lynx_28g_get_lanephy_node_optional(struct device *dev
 
 		if (val == lane)
 			return child;
-
-		of_node_put(child);
 	}
 
 	return NULL;
@@ -850,6 +848,7 @@ static int lynx_28g_probe(struct platform_device *pdev)
 			device_property_read_u32(&phy->dev, "fsl,eq-lnatecr0-25gbaser", &lane->eq_25gbaser.lnatecr0);
 			device_property_read_u32(&phy->dev, "fsl,eq-lnatecr1-25gbaser", &lane->eq_25gbaser.lnatecr1);
 		}
+		of_node_put(lane_np);
 
 		if (sysfs_create_group(&phy->dev.kobj, &lynx_28g_lane_group))
 			dev_warn(&pdev->dev, "failed to create sysfs group for lane %d\n", i);
-- 
2.51.2

